adding egg for counter-strike 2

Ballaual opened the pull request:
# Description

<!-- Please explain what is being changed or added as a short overview for this PR. Also, link existing relevant issues if they exist with resolves # -->

This PR adds a new egg for counter-strike 2. Due to the fact that there is no standalone branch like in csgo you will need to login with a steam account to download the game files. The installation might be a bit clunky when using steam guard (only mail works) so I added a detailed installation manual in the games README.md file. Also i added a template server.cfg file and download it from this repo when installing the server, because the default file is empty. I will maintain this egg as there will be any changes from Valve or they bring a standalone branch. Also i will add more content to this egg, e.g. Workshop Collections (if supported) over time. Any ideas or wishes are appreciated.

## Checklist for all submissions

<!-- insert X into the brackets to mark it as done (i.e. [x]). You can click preview to make the links appear clickable. -->

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

<!-- If this is an egg update fill these out -->

* [x] You verify that the start command applied does not use a shell script
  * [ ] If some script is needed then it is part of a current yolk or a PR to add one
* [x] The egg was exported from the panel

<!-- You can erase the new egg submission template if you're not adding a completely new egg -->

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
-> ghcr.io/pterodactyl/games:source    
* [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
    * [ ] If some script is needed then it is part of a current yolk or a PR to add one
6. [x] The egg was exported from the panel

QuintenQVD0 commented:
I am putting this on hold, as we do not like to run the client and would like to just wait for their dedicated server files. This will also not work as the client needs the sniper runtime installed in the docker image

QuintenQVD0 commented:
Also this needs the sniper runtime to be installed in the docker image

Ballaual commented:
> I am putting this on hold, as we do not like to run the client and would like to just wait for their dedicated server files.

I understand, i just wanted to share my attempt here, so anyone else can spin up a cs2 server if needed. If Valve decides to update the 740 depot working with cs2, i will update this egg immediately of course. But i dont really think that will happen in the near future as they need to fix their game first ;)
In the meantime i am working on some new stuff to be added and improving the install process.

With the latest [commit](https://github.com/parkervcp/eggs/pull/2490/commits/f9b17b2d1ecc26aa61e4812713244072344ffeb6) i added a new variable to determine which gamemode the server should start with. you can choose between, comp (default), wingman, casual, deathmatch and the custom gamemode.

I also cleaned up the startcommand and removed unnecessary parameters.
-usercon does not work either and rcon also working fine without it.
-console has no impact too, so i removed it

> Also this needs the sniper runtime to be installed in the docker image

This is a false information. The sniper runtime gets automatically installed by steamcmd when installing the 730 depot.
Installing or upgrading a game that runs in the sniper environment will download this compatibility tool `1628350` automatically. [Source](https://gitlab.steamos.cloud/steamrt/steamrt/-/blob/steamrt/sniper/README.md#where-can-i-get-it)
I just installed a fresh server and i was able to set it up as described the README, run it and able to join it without any further dependencies.

gOOvER commented:
> > I am putting this on hold, as we do not like to run the client and would like to just wait for their dedicated server files.
> 
> I understand, i just wanted to share my attempt here, so anyone else can spin up a cs2 server if needed. If Valve decides to update the 740 depot working with cs2, i will update this egg immediately of course. But i dont really think that will happen in the near future as they need to fix their game first ;) In the meantime i am working on some new stuff to be added and improving the install process.
> 
> With the latest [commit](https://github.com/parkervcp/eggs/pull/2490/commits/f9b17b2d1ecc26aa61e4812713244072344ffeb6) i added a new variable to determine which gamemode the server should start with. you can choose between, comp (default), wingman, casual, deathmatch and the custom gamemode.
> 
> I also cleaned up the startcommand and removed unnecessary parameters. -usercon does not work either and rcon also working fine without it. -console has no impact too, so i removed it
> 
> > Also this needs the sniper runtime to be installed in the docker image
> 
> This is a false information. The sniper runtime gets automatically installed by steamcmd when installing the 730 depot. Installing or upgrading a game that runs in the sniper environment will download this compatibility tool `1628350` automatically. [Source](https://gitlab.steamos.cloud/steamrt/steamrt/-/blob/steamrt/sniper/README.md#where-can-i-get-it) I just installed a fresh server and i was able to set it up as described the README, run it and able to join it without any further dependencies.

`Installing or upgrading a game that runs in the sniper environment will download this compatibility tool automatically. It can also be installed by running this command:`

You are wrong. It will NOT be installed by default. Only when game runs in Sniper Env

Ballaual commented:
> You are wrong. It will NOT be installed by default. Only when game runs in Sniper Env
> its NOT part of 730 Depot https://steamdb.info/depot/730/
I see, but it definetly also not correct that it needs to be installed separately, no? "Server" runs fine with default steamcmd install of 730.

> CS2 is atm more or lesser in a EA phase (Even though it was marked as a stable release). As long the biggest Bugs not fixed, there will never be a server released
That is absolutely correct, before game is not at least 99% fixed they wont touch the server files.

> And when it is released, it will definitely not be compatible with the client workaround and people post tons of support requests
Very sure it is not. I am not begging to force merge this ^^ as i said i will keep this updated as soon they release server files and in the meantime i will add more functionalities and QOL updates and will also update them with release of server files if needed.

Ballaual commented:
[329c60c Summary](https://github.com/parkervcp/eggs/pull/2490/commits/329c60c5f35592e105397fc40b50527d4be7772c)
- Adjusted variable name for user, password and steamguard matching to the names in the docker image `games:source`
-> STEAM_USER, STEAM_PASS, STEAM_AUTH so that the files can be updated on restart using the same steam user entered when installing the server. before was used anonymous automatically resulting in not correctly updating the files due to "no subscription" error.
- Ajusted the install script matching the changed variables
- Fix incorrect spaces between maps inside the SRCDS_MAP variable, resulting in not beeing able to select maps properly
- Added AUTO_UPDATE variable as boolean, with default value 1

Ballaual commented:
oh wow! Seems that valve brought a update for 740 app (server files) will test them and update the egg if working well!
https://steamdb.info/app/740/depots/

gOOvER commented:
ohh nice

QuintenQVD0 commented on game_eggs/steamcmd_servers/counter_strike/counter_strike_2/egg-counter--strike2.json:
```diff
@@ -22,8 +22,8 @@
-            "script": "#!\/bin\/bash\r\n# Installation Script\r\n\r\n## download and install steamcmd\r\ncd \/tmp\r\nmkdir -p \/mnt\/server\/steamcmd\r\ncurl -sSL -o steamcmd.tar.gz https:\/\/steamcdn-a.akamaihd.net\/client\/installer\/steamcmd_linux.tar.gz\r\ntar -xzvf steamcmd.tar.gz -C \/mnt\/server\/steamcmd\r\nmkdir -p \/mnt\/server\/steamapps # Fix steamcmd disk write error when this folder ...
-            "container": "ghcr.io\/pterodactyl\/installers:debian",
+            "script": "#!\/bin\/bash\r\n# steamcmd Base Installation Script\r\n#\r\n# Server Files: \/mnt\/server\r\n# Image to install with is 'ghcr.io\/parkervcp\/installers:debian'\r\n\r\n## just in case someone removed the defaults.\r\nif [[ \"${STEAM_USER}\" == \"\" ]] || [[ \"${STEAM_PASS}\" == \"\" ]]; then\r\n    echo -e \"steam user is not set.\\n\"\r\n    echo -e \"Using anonymous user. ...
+            "container": "ghcr.io\/parkervcp\/games:source",
```
Or is there a special reason for this?

Ballaual replied:
Don't ask me what got me into this... I actually wanted to use `ghcr.io/parkervcp/installers:debian`
Will patch it. Just found another mistake by me.. SteamGuard Codes are 5 digits not 6 digits and Backup Codes 7 digits not 8 digits.

Quick note:
As of now it does not look like that they will update `740` in the near future.
Latest update is meanwhile 18 days ago and targeted the legacy csgo version which will be supported at least until January 1st 2024. In addition to that the console asks now for gameserver login token which wasnt the last weks.

QuintenQVD0 reviewed (commented):
This looks fine to me but we prefere to wait until they release a dedicated app id for cs2.

Ballaual commented:
@QuintenQVD0 @gOOvER
Looks less and less like there will ever be standalone files for servers as they integrated 730 into anonymous now.
What do you think about that?

DylanDelobel commented:
> This looks fine to me but we prefere to wait until they release a dedicated app id for cs2.

They made CSGO available in a branch, pretty sure app id will not change for CS2

DylanDelobel commented:
What do we need to merge this? :)

QuintenQVD0 commented:
> What do we need to merge this? :)

The sniper runtime is still needed so may start by fixing that?

b-reich commented:
FYI: there is https://gitlab.steamos.cloud/steamrt/sniper/platform
